src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -32,6 +33,21 @@ internal class CSharpType
         {
         }
 
+        /// <summary>
+        /// Constructs a CSharpType for a literal type.
+        /// </summary>
+        /// <param name="type">The type to convert.</param>
+        /// <param name="isNullable">Flag used to determine if a type is nullable.</param>
+        /// <param name="isLiteral">Flag used to determine if a type is a literal.</param>
+        /// <param name="literalValue">The value of the literal.</param>
+        public CSharpType(Type type, bool isNullable, bool isLiteral, Constant? literalValue) : this(
```

Comment:
Can we remove the `isLiteral` we wouldn't want someone to set this to false but still pass in a `literalValue`

Reply:
Sure, will remove.